src/main/java/org/opensearch/ad/task/ADTaskCacheManager.java

```diff
@@ -681,6 +699,7 @@ public synchronized int scaleDownHCDetectorTaskSlots(String detectorId, int delt
         if (adTaskSlotLimit != null && delta > 0) {
             int newTaskSlots = taskSlots - delta;
             if (newTaskSlots > 0) {
+                taskSlots = newTaskSlots;
```

Comment:
why do we need this line? after this, newTaskSlots and taskSlots are equal and the log the next time does not seem to make sense.

Reply:
This is to return the new task slots. Yes ,the log will not show correct old task slots, will move this line after the log line 703.